libopae/plugins/xfpga/sysfs.c

```diff
@@ -314,16 +329,33 @@ void sysfs_foreach_region(region_cb cb, void *context)
 
 int sysfs_initialize(void)
 {
+	pthread_mutexattr_t mattr;
 	int stat_res = -1;
 	int reg_res = -1;
-	int res = 0;
+	int res = FPGA_OK;
 	uint32_t i = 0;
 	struct stat st;
 	DIR *dir = NULL;
 	char err[128] = {0};
 	struct dirent *dirent = NULL;
 	regex_t region_re;
 	regmatch_t matches[SYSFS_MAX_REGIONS];
+	if (pthread_mutexattr_init(&mattr)) {
```

Comment:
343-357 isn't necessary, because you've already init'ed the mutex with PTHREAD_RECURSIVE_MUTEX_INITIALIZER_NP.  That's the beauty of the static initializer.  Also, this code is prone to race conditions itself (because the lock is not held.)  Better to remove it.